dhcp-controller/api/v1alpha1/dhcpserver_types.go

```diff
@@ -22,18 +22,42 @@ import (
 
 // DHCPServerSpec defines the desired state of DHCPServer
 type DHCPServerSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Details of Servers
+	// +kubebuilder:validation:Required
+	Servers []Server `json:"servers,omitempty"`
+}
 
+type Server struct {
 	// refers to net-attach-def to be served
 	// +kubebuilder:validation:Required
 	NetworkName string `json:"networkName,omitempty"`
-	// refers to IP address to be configured at BindInterface port
+	// refers to IP address to bind interface to
 	// +kubebuilder:validation:Required
 	InterfaceIp string `json:"interfaceIp,omitempty"`
-	// configmap to be mounted as dnsmasq.conf
+	// refers to CIDR of server
+	// +kubebuilder:validation:Required
+	ServerCIDR CIDR `json:"cidr,omitempty"`
+	// refers to leasetime of IP
+	// +kubebuilder:validation:Required
+	LeaseTime string `json:"leaseTime,omitempty"`
+	// refers to vlan
+	VlanID string `json:"vlanid,omitempty"`
+}
+
+// CIDR defines CIDR of each network
+type CIDR struct {
+	// refers to start IP of range
+	// +kubebuilder:validation:Required
+	RangeStartIp string `json:"rangeStartIp,omitempty"`
+	// refers to end IP of range
+	// +kubebuilder:validation:Required
+	RangeEndIp string `json:"rangeEndIp,omitempty"`
+	// refers to subnet mask
+	// +kubebuilder:validation:Required
+	RangeNetMask string `json:"rangeNetMask,omitempty"`
+	// refers to gateway IP
 	// +kubebuilder:validation:Required
-	ConfigMapName string `json:"configMapName,omitempty"`
```

Comment:
Should we keep the config map? I think for advanced users if they want to supply more options or something we don't expose. SO we either create configmap OR we use the user-supplied one.

Reply:
We are still generating a configmap from the information provided in DHCPServer. So my thoughts were that if a user wishes to use additional configurations, then all they have to do is make a configmap with the same name as that of the DHCPServer. The dnsmasq pod should then redeploy with the updated config (yet to be implemented).

Reply:
Ok, so the configMap data gets merged/patched - what user supplies + what we generate internally?

Reply:
My thoughts were either what we generate or the complete custom one, but merging them makes more sense. Lesser that could go wrong.

Reply:
With Merging, I m not sure wether we are just complicating things,
so configmap option is for advanced users who want to try more option, In that case I think its better to leave it completely to administrator. So its either generated if not supplied or supplied.